packages/tizen_rpc_port/tizen/src/rpc_port_proxy.cc

```diff
@@ -32,37 +33,29 @@ flutter::EncodableMap CreateEncodableMap(const char* event,
 
 }  // namespace
 
-std::unique_ptr<FlEventSink> RpcPortProxyManager::event_sink_;
+std::map<rpc_port_proxy_h, std::unique_ptr<FlEventSink>>
+    RpcPortProxyManager::event_sinks_;
+
+void RpcPortProxyManager::Init(rpc_port_proxy_h handle,
+                               std::unique_ptr<FlEventSink> event_sink) {
+  auto it = event_sinks_.find(handle);
+  if (it == event_sinks_.end()) {
+    rpc_port_proxy_add_connected_event_cb(handle, OnConnectedEvent, handle);
+    rpc_port_proxy_add_disconnected_event_cb(handle, OnDisconnectedEvent,
+                                             handle);
+    rpc_port_proxy_add_rejected_event_cb(handle, OnRejectedEvent, handle);
+    rpc_port_proxy_add_received_event_cb(handle, OnReceivedEvent, handle);
+  }
 
-void RpcPortProxyManager::Init(std::unique_ptr<FlEventSink> event_sink) {
-  event_sink_ = std::move(event_sink);
+  event_sinks_[handle] = std::move(event_sink);
```

Comment:
What happens to the existing `event_sink` if another `FlStreamHandler` is registered to the same `handle`?

Reply:
I understand the intention of your question.
If it is common, it is correct to remove it when an `OnDisconnectedEvent` event occurs.
However, there are cases where connect is attempted again after disconnecting with the same handle.
I keep event_sink handle to recognize this. Should I remove this explicitly?
I naturally expect that the previous event_sink handle will call the annihilator as this is replaced.